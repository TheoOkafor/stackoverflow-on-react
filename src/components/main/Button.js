import PropTypes from 'prop-types';
import React from 'react';

const Button = (props) => {
  return (
    <button className={`btn ${props.styleName}`}
      id={props.id}
      value={props.value}>
      {props.name || props.children}
    </button>
  );
};

Button.propTypes = {
  styleName: PropTypes.string,
  id: PropTypes.string,
  children: PropTypes.element,
  value: PropTypes.string,
  name: PropTypes.string,
};

export default Button;
