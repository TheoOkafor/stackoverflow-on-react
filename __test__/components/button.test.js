import React from 'react';
import Button from '../../src/components/main/Button';
import renderer from 'react-test-renderer';

test('Test that the button displays', () => {
  const component = renderer.create(
    <Button styleName="primary" value="My Component" />,
  );
  expect(component).toMatchSnapshot();
});
