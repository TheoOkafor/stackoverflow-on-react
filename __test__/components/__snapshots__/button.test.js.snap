// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the button displays 1`] = `
<button
  className="btn primary"
  value="My Component"
/>
`;
